### PR TITLE
serial_no_listening: writing to a virtio serial port while no listening side

### PR DESCRIPTION
**Pull request by lijinlijin**
ID: 1745383

Signed-off-by: Li Jin <lijin@redhat.com>

### CODE REVIEW

**Comment by lijinlijin** [comment]
(1/2) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.virtio_serial_file_transfer.unix_socket.serial_no_listening: PASS (140.01 s)
 (2/2) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_serial_file_transfer.unix_socket.serial_no_listening: PASS (171.64 s)

@nanliu-r @xiagao 
Call for your review, thanks

File: qemu/tests/serial_no_listening.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+    # guest send while no listening side
+    try:
+        session.cmd_output(get_sender_cmd('guest'), timeout=timeout)
+    except aexpect.ShellTimeoutError:
+        pass
+
+    # host send while no listening side
+    try:
+        process.system(get_sender_cmd('host'), shell=True, timeout=timeout)
```
Comment (lolyu) [inline qemu/tests/serial_no_listening.py]: this `try` block is redundant, please use `ignore_status=True` instead.

**Comment by nanliu-r** [comment]
Test pass,thanks. 

(1/1) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.virtio_serial_file_transfer.unix_socket.serial_no_listening.q35: PASS (122.56 s)

**Comment by lijinlijin** [comment]
Modified, thanks. 
(1/2) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.virtio_serial_file_transfer.unix_socket.serial_no_listening: PASS (138.98 s)
 (2/2) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_serial_file_transfer.unix_socket.serial_no_listening: PASS (168.26 s)

File: qemu/tests/serial_no_listening.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            vm, port_name)
+
+    # guest send while no listening side
+    try:
+        session.cmd_output(get_sender_cmd('guest'), timeout=timeout)
+    except aexpect.ShellTimeoutError:
+        pass
+
+    # host send while no listening side
+    process.system(get_sender_cmd('host'), shell=True, timeout=timeout, ignore_status=True)
```
Comment (vivianQizhu) [inline qemu/tests/serial_no_listening.py]: You don't really check sending action here while you expect it fail, right?

File: qemu/tests/serial_no_listening.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                                                                test, driver)
+    virtio_serial_file_transfer.copy_scripts(guest_scripts, guest_path, vm)
+    port_type, port_path = virtio_serial_file_transfer.get_virtio_port_property(
+            vm, port_name)
+
+    # guest send while no listening side
+    try:
+        session.cmd_output(get_sender_cmd('guest'), timeout=timeout)
+    except aexpect.ShellTimeoutError:
+        pass
```
Comment (vivianQizhu) [inline qemu/tests/serial_no_listening.py]: Better have an `else` here, if you does not catch the exception, that's a fail, we don't expect that.

**Comment by vivianQizhu** [comment]
And better clarify your case name like `serial_no_listening` instead of `New case` in the commit title.

**Comment by lijinlijin** [comment]
@vivianQizhu 
I drop the use of serial data transfer use in this case, pls help to review, thanks.

 (1/2) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.serial_no_listening: PASS (183.01 s)
 (2/2) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.serial_no_listening: PASS (147.07 s)

**Comment by vivianQizhu** [comment]
@nanliu-r Call for your review, thanks.

File: qemu/tests/serial_no_listening.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    error_context.context("guest send while no listening side", logging.info)
+    try:
+        output = session.cmd_output(guest_send_cmd)
+    except aexpect.ShellTimeoutError:
+        if os_type != "linux":
+            test.error("timeout when guest send command:  %s" % guest_send_cmd)
+    else:
+        if not (os_type == "windows" and
+                ("The system cannot write to the specified device" in output)):
+            test.fail("Guest send should fail while no listening side")
```
Comment (nanliu-r) [inline qemu/tests/serial_no_listening.py]: We need to check guest at last, like operation `verify_kernel_crash`.

Comment (lijinlijin) [reply qemu/tests/serial_no_listening.py]: Updated, thanks

**Comment by nanliu-r** [comment]
Test pass, thanks.
ACK.

 (1/1) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.serial_no_listening.q35: PASS (170.18 s)

File: qemu/tests/serial_no_listening.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
+import logging
+import aexpect
+
+from avocado.utils import process
+
+from virttest import error_context
+from virttest import utils_test
+from qemu.tests import virtio_serial_file_transfer
```
Comment (vivianQizhu) [inline qemu/tests/serial_no_listening.py]: I would recommend only import `get_virtio_port_property` here.

Comment (lijinlijin) [reply qemu/tests/serial_no_listening.py]: Modified, thanks

**Pull request merged**